ignore static directory trailing slashes

Plasmoxy opened the pull request:
adds functionality that ignores trailing slashes when accessing a static directory ( asking for index.html )
for example if subdir with index.html is a directory in static file dir then :

**see** https://github.com/tipsy/javalin/issues/294

`http://localhost/subdir/` will return the subdir/index.html and `http://localhost/subdir` **will also return** subdir/index.html

you can also disable this functionality on static directories by calling `dontIgnoreTrailingStaticDirectorySlashes()` before server start(). ( this is similar to `dontIgnoreTrailingSlashes()` which **works for Javalin custom mapping but not for static directories** )

accessing files works as normal for example `http://localhost/file` will return the file content (if it is not a directory) and `http://localhost/file/` will try to access it as directory (resulting in 404 if its a file ).
( which is logical )

check https://github.com/Plasmoxy/javalin/blob/2068c6dce74b112ed3bb47ee333ef23702991918/src/test/java/io/javalin/TestStaticDirectorySlash.kt for tests

Plasmoxy commented:
also I hope i didn't break travis by putting a while(true) { Thread.sleep(1000) } inside a test ( https://github.com/tipsy/javalin/blob/2068c6dce74b112ed3bb47ee333ef23702991918/src/test/java/io/javalin/misc/StaticDirectorySlashRuntimeTest.kt ), please excuse me for this, this is my first time using JUnit or travis :/

tipsy commented:
Thanks @Plasmoxy, nice work!

I agree with what @pkkummermo that `dontIgnoreTrailingSlashes()` should be used for this., meaning the new behavior should be the default, and users can to disable it by calling `dontIgnoreTrailingSlashes()`.

> also I hope i didn't break travis 

Travis doesn't mind being broken, pushing a working commit will always unbreak him.

> please excuse me for this, this is my first time using JUnit or travis :/

Very good job then!

Plasmoxy commented:
Thank you @tipsy ! I'm happy that I can contribute to Javalin ;) ( want to use it for production too ). I've added the `dontIgnoreTrailingStaticDirectorySlashes()` functionality to `dontIgnoreTrailingSlashes()` , removed my function, fixed tests.

tipsy commented:
> Thank you @tipsy ! I'm happy that I can contribute to Javalin ;) ( want to use it for production too ).

Thank you too, contributions are very much appreciated. You can feel comfortable using it in production, all the hard technical operations are taken care of by [Jetty](https://github.com/eclipse/jetty.project), which is an extremely well run project.
